miner/worker.go

```diff
@@ -396,7 +396,7 @@ func (miner *Miner) commitBlobTransaction(env *environment, tx *types.Transactio
 	// isn't really a better place right now. The blob gas limit is checked at block validation time
```

Comment:
@qizhou suggests addressing this case at line 392 to make the code self-explanatory and explicitly handle all possible scenarios.
```suggestion
	nblobs := 0
	if sc == nil {
		if env.isEffectivelySequencing() /* we want to allow blob tx without blobs when it's deriving */ {
			panic("blob transaction without blobs in sequencing")
		} else { // deriving, which does not have the sidecar
			nblobs = len(tx.BlobHashes())
		}
	} else {
		if !env.isEffectivelySequencing() {
			panic("blob transaction with blobs in derivation")
		}
		nblobs = len(sc.Blobs)
	}
	// should we check tx.BlobHashes() matches that of sc.Blobs.BlobHashes()?
	// Checking against blob gas limit: It's kind of ugly to perform this check here, but there
	// isn't really a better place right now. The blob gas limit is checked at block validation time
	// and not during execution. This means core.ApplyTransaction will not return an error if the
	// tx has too many blobs. So we have to explicitly check it here.
	if (env.blobs+nblobs)*params.BlobTxBlobGasPerBlob > params.MaxBlobGasPerBlock {
		return errors.New("max data blobs reached")
	}	
```

Reply:
Done in 80982988ab9aacd190523c0aa266fe61080b3ecf.